src/ReduxWPAPI.js

```diff
@@ -41,151 +37,35 @@ export default class ReduxWPAPI {
-      const foundIndexers = reduce(request._path, (indexers, fragment, piece) => {
-        const id = find(request._levels[piece], cmp => cmp.validate(fragment));
-
-        if (!id) return indexers;
-        const name = id.component.match(namedGroupRegex);
-
-        if (name) {
-          lastFragment = indexers[name[1]] = fragment;
-          unresolvedNesting--;
-        } else {
-          unresolvedNesting++;
-        }
-
-        return indexers;
-      }, { ...request._params });
-
-      if (!foundIndexers.id && lastFragment && unresolvedNesting === 1) {
-        foundIndexers.id = lastFragment;
-      }
-
-      return foundIndexers;
-    },
-
-    getRequestUID(request, { api }) {
-      /* eslint-disable no-param-reassign */
-      // Internal mutations are reversed before return
-      const page = request._params.page;
-      const _embed = request._params._embed;
-      delete request._params.page;
-      delete request._params._embed;
-
-      const uid = request._renderURI();
-      request._params.page = page;
-      request._params._embed = _embed;
-
-      return uid.replace(api._options.endpoint, '');
-    },
-
-    embed(_embedded, toEmbed, link, relName) {
-      let linkRenamed = relName.replace(/^https:\/\/api\.w\.org\//, '');
-
-      switch (linkRenamed) {
-        case 'featuredmedia': linkRenamed = 'featured_media'; break;
-        case 'term':
-          linkRenamed = link.href.replace(/\?.*$/, '').replace(/.*\/([^\/]+)$/, '$1');
-          break;
-        default:
-      }
-
-      return { ..._embedded, [linkRenamed]: toEmbed };
-    },
-
-    getUrl(request) {
-      return request._renderURI();
-    },
-
-    getAggregator(url, { api }) {
-      let uri = url.replace(api._options.endpoint, '').replace(/\?.*$/, '');
-      const namespace = findKey(api._ns, (factory, ns) => uri.indexOf(ns) === 0);
-
-      if (!namespace) return null;
-
-      uri = uri.replace(`${namespace}/`, '');
-      const fragments = uri.split('/');
-      const [resource] = fragments;
-      const query = api[resource]();
-      let aggregator = resource;
-
-      for (let piece = 1; piece < fragments.length; piece++) {
-        const id = find(query._levels[piece], cmp => cmp.validate(fragments[piece]));
-
-        if (!id) {
-          return false;
-        }
-
-        if (!id.component.match(namedGroupRegex)) {
-          aggregator += capitalize(id.component);
-        }
-      }
-
-      if (aggregator === 'usersMe') return 'users';
-
-      return aggregator;
-    },
-
-    callAPI(request, operation, params) {
-      return request.embed()[operation](params);
-    },
   }
 
   indexers = {}
 
   constructor(settings) {
     this.settings = defaultsDeep({}, settings, ReduxWPAPI.defaultSettings);
 
-    if (!this.settings.api) {
-      throw new Error(`${ReduxWPAPI.displayName}: 'api client must be provided`);
+    if (!this.settings.adapter) {
+      this.settings.adapter = new WPAPIAdapter(this.settings);
```

Comment:
Don't really like this. It feels like the fish is eating itself. How about having the adapter in the same level?

`this.adapter = new WPAPIAdapter(this.settings)`